homeassistant/components/idasen_desk/__init__.py

```diff
@@ -68,20 +70,24 @@ async def async_disconnect(self) -> None:
         self._connection_lost = False
         await self.desk.disconnect()
 
-    @callback
-    def async_set_updated_data(self, data: int | None) -> None:
-        """Handle data update."""
+    async def async_ensure_connection_state(self) -> None:
+        """Check if the expected connection state matches the current state and correct it if needed."""
         if self._expected_connected:
             if not self.desk.is_connected:
-                _LOGGER.debug("Desk disconnected. Reconnecting")
+                _LOGGER.warning("Desk disconnected. Reconnecting")
```

Comment:
```suggestion
                _LOGGER.debug("Desk disconnected. Reconnecting")
```

Let's keep this at debug since its not something the user can do anything about